src/js/application.js

```diff
@@ -0,0 +1,288 @@
+        this.language = 'en';
+
+        this.currentLocation;
+        this.currentWeather;
+    }
+
+    init() { 
+        this.initActiveDegreeButton();
+        const SEARCH_LOCATION_ELEMENT = document.querySelector(selectors.SEARCH_FORM);
+        SEARCH_LOCATION_ELEMENT.addEventListener('submit', this.onLocationChange.bind(this));  
+
+        const LANGUAGE_BUTTON_ELEMENT = document.querySelector(selectors.LANGUAGE_BTN);
+        LANGUAGE_BUTTON_ELEMENT.addEventListener('change', this.changeLanguage.bind(this)); 
+
+        const BUTTONS_CONTAINER = document.querySelector(selectors.BUTTONS_CONTAINER);
+        BUTTONS_CONTAINER.addEventListener('click', this.determineClickedButton.bind(this));
+        
+        this.changeBackground();
+        const savedLanguage = localStorage.getItem('language');
+        if (savedLanguage && savedLanguage != this.language) {
+            this.setLanguage(savedLanguage, false);
+        }
+        this.loadLocationByIP();
+    }
+
+    showErrorMessage() {
+        const ERROR_MESSAGE_ELEMENT = document.querySelector(selectors.ERROR_MESSAGE);
+        ERROR_MESSAGE_ELEMENT.style.display = 'block';
+    
+        setTimeout(() => {
+            ERROR_MESSAGE_ELEMENT.style.display = 'none';
+        }, 3000);    
+    }
+
+    async onLocationChange(e) {    
+        e.preventDefault();
+
+        const SEARCH_INPUT_ELEMENT = document.querySelector(selectors.SEARCH_INPUT_FIELD);
+        const location = await Location.getLocationByName(SEARCH_INPUT_ELEMENT.value, this.language);
+
+        if (location) {
+            await location.loadLocationTimezone();
+            await this.setLocation(location);
+        } else {
+            this.showErrorMessage();
+        }
+    }
+
+    showDate() {
+        const DATE_ELEMENT = document.querySelector(selectors.DATE);
+
+        DATE_ELEMENT.innerHTML = this.currentLocation.getDate();
+    }
+
+    showTime() {
+        const TIME_ELEMENT = document.querySelector(selectors.TIME);
+
+        TIME_ELEMENT.innerHTML = this.currentLocation.getTime();
+        setTimeout(this.showTime.bind(this), 1000);
+    }
+
+    async loadLocationByIP() {
+        let response = await fetch('https://ipinfo.io/json?token=edaa4c28dad526');
+
+        response = await response.json();
+        const location = await Location.getLocationByName(response.city, this.language);
+        if (location) {
+            await this.setLocation(location);
+        }
+    }
+
+    async setLocation(location) {
+        const LATITUDE_DEGREES_ELEMENT = document.querySelector(selectors.LATITUDE_DEGREES);
+        const LONGITUDE_DEGREES_ELEMENT = document.querySelector(selectors.LONGITUDE_DEGREES);
+        const LATITUDE_MINUTES_ELEMENT = document.querySelector(selectors.LATITUDE_MINUTES);
+        const LONGITUDE_MINUTES_ELEMENT = document.querySelector(selectors.LONGITUDE_MINUTES);
+        const LOCATION_ELEMENT = document.querySelector(selectors.LOCATION);
```

Comment:
too many logic at one place
at least try to move updating of innerHTML  to a separate method

Reply:
fixed